src/test/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/action/ForceMergeActionIT.kt

```diff
@@ -0,0 +1,222 @@
+        insertSampleData(indexName, 3, 1000)
+
+        val segmentCount = getSegmentCount(indexName)
+        assertTrue("Segment count for [$indexName] was less than expected", segmentCount > 1)
+
+        // Add policy to index after data is loaded
+        addPolicyToIndex(indexName, policyID)
+        Thread.sleep(2000)
+
+        val managedIndexConfig = getManagedIndexConfig(indexName)
+        assertNotNull("ManagedIndexConfig is null", managedIndexConfig)
+
+        // Will change the startTime each execution so that it triggers in 2 seconds
+        // First execution: Policy is initialized
+        updateManagedIndexConfigStartTime(managedIndexConfig!!, Instant.now().minusSeconds(58).toEpochMilli())
+        Thread.sleep(3000)
+
+        // Second execution: Index is set to read-only for force_merge
+        updateManagedIndexConfigStartTime(managedIndexConfig, Instant.now().minusSeconds(58).toEpochMilli())
+        Thread.sleep(3000)
+
+        assertEquals("true", getIndexBlocksWriteSetting(indexName))
+
+        // Third execution: Force merge operation is kicked off
+        updateManagedIndexConfigStartTime(managedIndexConfig, Instant.now().minusSeconds(58).toEpochMilli())
+        Thread.sleep(3000)
+
+        // Fourth execution: Waits for force merge to complete, which will happen in this execution since index is small
+        updateManagedIndexConfigStartTime(managedIndexConfig, Instant.now().minusSeconds(58).toEpochMilli())
+        Thread.sleep(3000)
+
+        val segmentCountAfterForceMerge = getSegmentCount(indexName)
+        assertEquals("Segment count for [$indexName] after force merge is incorrect", 1, segmentCountAfterForceMerge)
+
+        // Fifth execution: Set index back to read-write since it was not originally read-only
+        updateManagedIndexConfigStartTime(managedIndexConfig, Instant.now().minusSeconds(58).toEpochMilli())
+        Thread.sleep(3000)
+
+        assertEquals("false", getIndexBlocksWriteSetting(indexName))
+
+        // Sixth execution: Transition to DeleteState
+        updateManagedIndexConfigStartTime(managedIndexConfig, Instant.now().minusSeconds(58).toEpochMilli())
+        Thread.sleep(3000)
+
+        // Seventh execution: Delete the index
+        updateManagedIndexConfigStartTime(managedIndexConfig, Instant.now().minusSeconds(58).toEpochMilli())
+        Thread.sleep(3000)
+
+        assertFalse("Index [$indexName] was not deleted", indexExists(indexName))
+    }
+
+    fun `test force merge on index already in read-only`() {
+        val indexName = "${testIndexName}_index"
+        val policyID = "${testIndexName}_testPolicyName"
+
+        // Create a Policy with two States, one that set index to read-only and one that force-merges
+        val readOnlyActionConfig = ReadOnlyActionConfig(index = 0)
+        val forceMergeActionConfig = ForceMergeActionConfig(maxNumSegments = 1, index = 0)
+        val states = listOf(
+            State(
```

Comment:
Similar to above, you could just manually set the index to read-only instead of relying on a state/action transition which is adding time to the overall testing time. We already test readonly actions in its own IT.